internal/config/config.go

```diff
@@ -47,28 +77,36 @@ func Read(config *Configuration, configFile string) (found bool, configFileUsed
 	v.SetDefault("Worker.MaxConcurrentSessions", 1)
 
 	if configFile != "" {
+		// Viper will not return a viper.ConfigFileNotFoundError error when
+		// SetConfigFile() is passed a path to a file that doesn't exist, so we
+		// need to check ourselves.
+		if _, err := os.Stat(configFile); errors.Is(err, os.ErrNotExist) {
+			return false, "", fmt.Errorf("configuration file not found: %s", configFile)
+		}
```

Comment:
It will not error when `SetConfigFile` is used, but it will error below when `ReadInConfig` is called with the `viper.ConfigFileNotFoundError` you are checking, or is that a different error?

Reply:
@jraddaoui yeah, I think it should be considered a bug with viper:

- If `SetConfigFile()` is called, then viper just blindly sets the path to passed string and when `viper.Unmarshal()` is called it throws a generic viper error (not a `viper.ConfigFileNotFoundError` as I would expect).
- If `SetConfigFile()` is *not* called then viper will search for a config file in all of the `v.AddConfigPath()` paths, and if it doesn't find a config file in any of them it will return a viper.ConfigFileNotFoundError`.

Reply:
Ouch, I see, thanks for the explanation @djjuhasz!